contracts/integrations/test/dydx_integration_test.ts

```diff
@@ -0,0 +1,441 @@
+        await addBalanceAndAllowanceAsync(weth, untrustedSender, deploymentManager.dydx.exchangeWrapper.address);
+        await addBalanceAndAllowanceAsync(weth, tradeOriginator, deploymentManager.dydx.exchangeWrapper.address);
+
+        // The exchange wrapper needs a takerToken balance, but isn't an EOA
+        await takerToken.setBalance.awaitTransactionSuccessAsync(owner, constants.INITIAL_ERC20_BALANCE, {
+            from: owner,
+        });
+        await takerToken.transfer.awaitTransactionSuccessAsync(
+            deploymentManager.dydx.exchangeWrapper.address,
+            constants.INITIAL_ERC20_BALANCE,
+            { from: owner },
+        );
+    });
+
+    describe('ZeroExV3ExchangeWrapper', () => {
+        function dydxOrderData(order: SignedOrder): string {
+            return '0x'.concat(
+                Buffer.concat(
+                    [
+                        order.makerAddress,
+                        order.takerAddress,
+                        order.feeRecipientAddress,
+                        order.senderAddress,
+                        order.makerAssetAmount,
+                        order.takerAssetAmount,
+                        order.makerFee,
+                        order.takerFee,
+                        order.expirationTimeSeconds,
+                        order.salt,
+                        makerFeeToken.address,
+                        takerFeeToken.address,
+                    ].map(field => ethUtil.setLengthLeft(toHex(field), constants.WORD_LENGTH)),
+                ).toString('hex'),
+                order.signature.substr(2),
+            );
+        }
+
+        async function getBalancesAsync(): Promise<TokenBalances> {
+            const addresses = [
+                deploymentManager.dydx.exchangeWrapper.address,
+                deploymentManager.staking.stakingProxy.address,
+                ...accounts,
+            ];
+            const tokens = [makerToken, takerToken, makerFeeToken, takerFeeToken, weth];
+            const erc20Balances = _.zipObject(
+                addresses,
+                await Promise.all(
+                    addresses.map(async account =>
+                        _.zipObject(
+                            tokens.map(token => token.address),
+                            await Promise.all(tokens.map(token => token.balanceOf.callAsync(account))),
+                        ),
+                    ),
+                ),
+            );
+            const ethBalances = _.zipObject(
+                addresses,
+                await Promise.all(addresses.map(address => env.web3Wrapper.getBalanceInWeiAsync(address))),
+            );
+
+            return {
+                erc20: erc20Balances,
+                erc721: {},
+                erc1155: {},
+                eth: ethBalances,
+            };
+        }
+
+        function getExpectedBalances(
```

Comment:
This balance checking stuff really should be DRYed up; I'm working on a PR to use `BalanceStore` in more places, which will address this.